backend/trpc/index.ts

```diff
@@ -0,0 +1,525 @@
+          message: `Failed to update notebook: ${error instanceof Error ? error.message : "Unknown error"}`,
+        });
+      }
+    }),
+
+  // Delete notebook
+  deleteNotebook: authedProcedure
+    .input(z.object({ nbId: z.string() }))
+    .mutation(async (opts) => {
+      const { ctx, input } = opts;
+      const { nbId } = input;
+      const {
+        user,
+        env: { DB },
+        permissionsProvider,
+      } = ctx;
+
+      try {
+        // Check if user is owner
+        const isOwner = await permissionsProvider.isOwner(user.id, nbId);
+        if (!isOwner) {
+          throw new TRPCError({
+            code: "FORBIDDEN",
+            message: "Only the owner can delete a notebook",
+          });
+        }
+
+        // Delete notebook (CASCADE will handle permissions)
+        const result = await DB.prepare("DELETE FROM notebooks WHERE id = ?")
+          .bind(nbId)
+          .run();
+
+        if (result.meta.changes === 0) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Notebook not found",
+          });
+        }
+
+        return true;
+      } catch (error) {
+        if (error instanceof TRPCError) throw error;
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: `Failed to delete notebook: ${error instanceof Error ? error.message : "Unknown error"}`,
+        });
+      }
+    }),
+
+  // Share notebook
+  shareNotebook: authedProcedure
+    .input(
+      z.object({
+        nbId: z.string(),
+        userId: z.string(),
+      })
+    )
+    .mutation(async (opts) => {
+      const { ctx, input } = opts;
+      const { user, permissionsProvider } = ctx;
+
+      try {
+        await permissionsProvider.grantPermission({
+          notebookId: input.nbId,
+          userId: input.userId,
+          grantedBy: user.id,
+        });
+
+        return true;
+      } catch (error) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: `Failed to share notebook: ${error instanceof Error ? error.message : "Unknown error"}`,
+        });
+      }
+    }),
+
+  // Unshare notebook
+  unshareNotebook: authedProcedure
+    .input(
+      z.object({
+        nbId: z.string(),
+        userId: z.string(),
+      })
+    )
+    .mutation(async (opts) => {
+      const { ctx, input } = opts;
+      const { user, permissionsProvider } = ctx;
+
+      try {
+        await permissionsProvider.revokePermission({
+          notebookId: input.nbId,
+          userId: input.userId,
+          revokedBy: user.id,
+        });
+
+        return true;
+      } catch (error) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: `Failed to unshare notebook: ${error instanceof Error ? error.message : "Unknown error"}`,
+        });
+      }
+    }),
+
+  // Get notebook owner
+  notebookOwner: authedProcedure
+    .input(z.object({ nbId: z.string() }))
```

Comment:
I like the flexibility here of being able to pass an on demand zod schema or create one above to pass in.